### PR TITLE
[Catalog] Provide app-wide MDCContainerScheme property

### LINKED ISSUE #6443: [MDCCatalog] Provide app-wide MDCContainerScheme property
This was filed as an internal issue. If you are a Googler, please visit [b/123234863](http://b/123234863) for more details.

---
#### Internal data
- Associated internal bug: [b/123234863](http://b/123234863)

### PR DESCRIPTION
**Pull request by andrewoverton**
My first thought was to add an MDCContainerScheme property to the app delegate, but then I remembered we had this AppTheme global and figured that was where it should go.

I don't really see how this change makes anything better, but then again maybe I'm doing it wrong?
🤔🙈

I haven't looked into whether the swift MaterialComponentsBeta imports will cause problems, but I can imagine them doing so...

Closes #6443.

### CODE REVIEW

**Comment by jverkoey** [comment]
How close are we able to get to theming the app with only the container scheme and theming extensions?

I agree that as it stands this PR looks a bit like moving bits around; I think the value here will be in consolidating the theming on the extension APIs, removing the workarounds to extract subsystem schemes from the container.

**Review by romoore — DISMISSED** [review]
This looks like much more than providing the scheme. Please split the actual theming calls into separate PRs.

File: catalog/MDCCatalog/AppTheme.swift [hunk 1]
```diff
@@ -16,22 +16,25 @@ import Foundation
 import MaterialComponents.MaterialButtons_ButtonThemer
 import MaterialComponents.MaterialColorScheme
 import MaterialComponents.MaterialTypographyScheme
+import MaterialComponentsBeta.MaterialContainerScheme
 
 final class AppTheme {
-  let colorScheme: MDCColorScheming
```
Comment (romoore) [inline catalog/MDCCatalog/AppTheme.swift:21]: At the moment, these properties can be left in-place to reduce required code changes. We can do gradual migrations if we simply assign the `colorScheme` property of the `containerScheme` to be the same as the `colorScheme` property in this class. Same for TypographyScheme and ShapeScheme.

**Comment by andrewoverton** [comment]
@romoore and @jverkoey -- Should this PR be a one line addition of the container scheme property in AppTheme? No other changes?

**Comment by jverkoey** [comment]
> Should this PR be a one line addition of the container scheme property in AppTheme? No other changes?

I think a good simple starting point would be to ensure the following are true:

1. There is a single container scheme in the app theme object.
2. That container scheme gets provided to example view controllers via the runtime setting logic (similar to how the individual subsystem schemes are being provided today).

I think it would be fair to do both of the above in a single change.

There would then be two categories of follow-up changes:

1. Updating the catalog to use the container scheme and Theming extension APIs where possible.
2. Updating each example to use the injected container scheme instead of the individual subsystem schemes.

**Comment by romoore** [comment]
@jverkoey's approach sounds good to me. The only improvement I would suggest is that the `colorScheme` and `containerScheme.colorScheme` objects are identical. The same for TypographyScheme (and Shape, but it doesn't seem to get injected).

**Comment by andrewoverton** [comment]
Okie dokie, so here's what I'll do:

1. Keep the container scheme property on AppTheme, but undo the colorScheme and typographyScheme deletions, as well as the code changes those deletions necessitated.

2. Add a `-setContainerScheme:` "if" clause [here](https://www.google.com/url?sa=j&url=https%3A%2F%2Fgithub.com%2Fmaterial-components%2Fmaterial-components-ios%2Fblob%2Fdevelop%2Fcatalog%2FMDCCatalog%2FMDCNodeListViewController.swift%23L589&uct=1525096300&usg=6ymx-EWiSXen8QkZAkKspUsuybA.&source=chat).

Thank you for clarifying!

**Review by romoore — APPROVED** [review]

**Review by jverkoey — APPROVED** [review]

**Comment by jverkoey** [comment]
Looks like the three failing CI jobs are unrelated to this PR, so carry on! I'm looking in to the website failures right now.

**Pull request merged**